ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/PaywallDataExtensions.kt

```diff
@@ -0,0 +1,100 @@
+package com.revenuecat.purchases.ui.revenuecatui.extensions
+
+import androidx.compose.material.Colors
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.paywalls.PaywallColor
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.PaywallTemplate
+import java.net.URL
+import java.util.Locale
+
+/***
+ * Default [PaywallData] to display when attempting to present a [PaywallView] with an offering that has no paywall
+ * configuration, or when that configuration is invalid.
+ */
+internal fun PaywallData.Companion.createDefault(
+    packages: List<Package>,
+    packageName: String,
+    currentColors: Colors,
+): PaywallData {
+    return PaywallData.createDefaultForIdentifiers(packages.map { it.identifier }, packageName, currentColors)
+}
+
+private fun PaywallData.Companion.createDefaultForIdentifiers(
+    packageIdentifiers: List<String>,
+    packageName: String,
+    currentColors: Colors,
+): PaywallData {
+    return PaywallData(
+        templateName = PaywallData.defaultTemplate.id,
+        config = PaywallData.Configuration(
+            packages = packageIdentifiers,
+            images = PaywallData.Configuration.Images(
+                background = PaywallData.defaultBackgroundImage,
+                icon = PaywallData.defaultAppIconPlaceholder,
+            ),
+            colors = PaywallData.defaultColors(currentColors),
+            blurredBackgroundImage = true,
+            displayRestorePurchases = true,
+        ),
+        localization = mapOf(Locale.ENGLISH.toString() to PaywallData.defaultLocalization),
+        assetBaseURL = PaywallData.defaultTemplateBaseURL(packageName),
+        revision = PaywallData.revisionID,
+    )
+}
+
+// region Internal defaults
+
+internal val PaywallData.Companion.defaultTemplate: PaywallTemplate
+    get() = PaywallTemplate.TEMPLATE_2
+
+internal val PaywallData.Companion.defaultAppIconPlaceholder: String
+    get() = "revenuecatui_default_paywall_app_icon"
+
+// endregion
+
+// region Private defaults
+
+private val PaywallData.Companion.revisionID: Int
+    get() = -1
+
+private val PaywallData.Companion.defaultBackgroundImage: String
+    get() = "R.drawable.default_background"
+
+private val PaywallData.Companion.defaultLocalization: PaywallData.LocalizedConfiguration
+    get() = PaywallData.LocalizedConfiguration(
+        title = "{{ app_name }}",
+        callToAction = "Continue",
+        offerDetails = "{{ total_price_and_per_month }}",
+        offerDetailsWithIntroOffer = "Start your {{ sub_offer_duration }} trial, then {{ total_price_and_per_month }}.",
+    )
+
+private fun PaywallData.Companion.defaultTemplateBaseURL(packageName: String): URL =
+    URL("android.resource://$packageName/")
```

Comment:
I tested this and it doesn't work, so I am just going to revert it to use a placeholder, same way we do in iOS. It requires less refactoring

Reply:
👍🏻

Reply:
Oh looks like you never reverted this. It's making the app crash right now.